en/docs/insights/choreo-insights-api.md

```diff
@@ -0,0 +1,4214 @@
+
+<table>
+<thead>
+<tr>
+<th align="left">Field</th>
+<th align="right">Argument</th>
+<th align="left">Type</th>
+<th align="left">Description</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td colspan="2" valign="top"><strong>timeSpan</strong></td>
+<td valign="top"><a href="#string">String</a></td>
+<td>
+
+Represents a timestamp value calculated according to the used granularity value. For an example if the used
+granularity value is '1d', timeSpan will represents start of the day calculated adhering to the timezone provided in
+the TimeFilter.
+Ex: timeSpan: '2021-06-21T00:00:00.0000000+05:30'
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>errorCount</strong></td>
+<td valign="top"><a href="#int">Int</a></td>
+<td>
+
+Represents error request count.
+
+</td>
+</tr>
+</tbody>
+</table>
+
+### ErrorsByCategory
+
+Represents errors by category over time for error categories.
+
+<table>
+<thead>
+<tr>
+<th align="left">Field</th>
+<th align="right">Argument</th>
+<th align="left">Type</th>
+<th align="left">Description</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td colspan="2" valign="top"><strong>errors</strong></td>
+<td valign="top">[<a href="#errorcountbycategory">ErrorCountByCategory</a>]</td>
+<td>
+
+List containing error count for each error category over time. Returns an empty array if no data available.
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>granularity</strong></td>
+<td valign="top"><a href="#string">String</a>!</td>
+<td>
+
+Granularity value used for data retrieval.
+
+</td>
+</tr>
+</tbody>
+</table>
+
+### ErrorsByStatusCode
+
+Represents error counts for each status code for each API for all the APIs.
+
+<table>
+<thead>
+<tr>
+<th align="left">Field</th>
+<th align="right">Argument</th>
+<th align="left">Type</th>
+<th align="left">Description</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td colspan="2" valign="top"><strong>errors</strong></td>
+<td valign="top">[<a href="#errorcountbycodeforapi">ErrorCountByCodeForAPI</a>]!</td>
+<td>
+
+List containing error count for each status code for each API. Returns an empty array if no data available.
+
+</td>
+</tr>
+<tr>
+<td colspan="2" valign="top"><strong>pagination</strong></td>
+<td valign="top"><a href="#pagination">Pagination</a>!</td>
+<td>
+
+Pagination details.
+
+</td>
+</tr>
+</tbody>
+</table>
+
+### ErrorsByStatusCodeCategory
+
+Represents error category values(as 4xx, 5xx and total) for both proxy and target errors for a single timestamp.
+Timestamp is calculated according to the provided granularity value.
+
+<table>
+<thead>
+<tr>
+<th align="left">Field</th>
+<th align="right">Argument</th>
+<th align="left">Type</th>
+<th align="left">Description</th>
+</tr>
+</thead>
+<tbody>
+<tr>
+<td colspan="2" valign="top"><strong>timeSpan</strong></td>
+<td valign="top"><a href="#string">String</a>!</td>
+<td>
+
+Represents a timestamp value calculated according to the used granularity value. For an example if the used
```

Comment:
When you say "used granularity value" it is not clear as to where it is used and by what. Please check whether the following suggestion gives the same meaning.

```suggestion
Represents a timestamp value that is calculated based on the specified granularity value. For example, if the granularity value is `1d`, the timestamp represents the start of the day calculated adhering to the timezone provided in the time filter.
e.g., timeSpan: '2021-06-21T00:00:00.0000000+05:30'
```